Adds Ctrl + Enter shortcut for expression evaluation

streamich opened the pull request:
## Summary

- This allows to execute queries by pressing `Ctrl + Enter` instead of having to mouse-click on "Run" button.
- I did not find tests for the query editor component, so didn't add any for this keyboard shortcut either.

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- ~~This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~~
- ~~Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~~
- ~~[Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~~
- [ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios
- ~~This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~~

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)
- [ ] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)

elasticmachine commented:
Pinging @elastic/kibana-canvas

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/12793/)

cqliu1 commented:
@streamich We've been meaning to work on this feature (#35874). Thanks for taking it on!

We use `react-shortcuts` to handle our keyboard shortcuts which requires a keymap config, and all of the shortcuts in Canvas are defined in [`/canvas/public/lib/keymap.js`](https://github.com/elastic/kibana/blob/dcf5f91784e320cd097d63b2df38622b03883e50/x-pack/plugins/canvas/public/lib/keymap.js#L57-L124), which we also use for generating the shortcuts reference available in the global nav help menu.

![Screen Shot 2019-05-06 at 1 55 14 PM](https://user-images.githubusercontent.com/1697105/57254630-9d23e200-7006-11e9-9e2a-70cbe2c3b9c5.png)

Could you add your shortcut to the `keymap` and include some help text as well?

cqliu1 commented:
Tested out your branch, and it's much nicer to be able to run the expression with a shortcut. 

I ran into one issue. It looks like you can run an invalid expression with `cmd/ctrl+enter` even if the `Run` button is disabled. Can you disable the shortcut when there's an error with the expression?

![May-06-2019 15-52-07](https://user-images.githubusercontent.com/1697105/57260097-13304500-7017-11e9-9546-30617548ab30.gif)

streamich commented:
@cqliu1 Thanks for reviewing, sorry I didn't know there is an issue for this. 

Addressed your comments:

- added shortcut to `keymap`
- using `react-shortcuts` lib to trigger events (interesting library, will try it out in my projects)
- also added `Shift + ...` for partial selection as in #35874, but I'm not sure if how it works is something you want.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/12889/)

cqliu1 commented:
Thanks for adding the shortcuts to the keymap and using `react-shortcuts`!

I would hold off on adding the shortcut for running the partial expression for now. We don't actually want to using `setExpression` to persist the partial expression to the stored element. We essentially want to run a selection to preview the output of the expression up to a certain point and would probably want a dedicated UI for the preview. That feature is out of the scope of this PR. Just having the shortcut to run the whole expression is huge improvement for now.

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/13180/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/14710/)
